docs/api/api.py

```diff
@@ -25,14 +25,16 @@ def local_git_repo(path: str) -> Repo:
 def docker_build(ref: str, context: str, build_args: Dict[str, str] = {}, dockerfile: Union[str, LocalPath] = "Dockerfile", dockerfile_contents: Union[str, Blob] = "") -> None:
   """Builds a docker image.
 
-  Note that you can't set both the dockerfile and dockerfile_contents arguments.
+  Note that you can't set both the `dockerfile` and `dockerfile_contents` arguments (will throw an error).
+
+  Example: ``docker_build('myregistry/myproj/backend', '/path/to/code')`` is roughly equivalent to the call ``docker build /path/to/code -t myregistry/myproj/backend`` (and will look for a Dockerfile at ``/path/to/code/Dockerfile``)
 
   Args:
-    ref: e.g. a blorgdev/backend or gcr.io/project-name/bucket-name
-    context: The path to use as the Docker build context.
-    build_args: the build-time variables that are accessed like regular environment variables in the ``RUN`` instruction of the Dockerfile. See `the Docker Build Arg documentation <https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg>`_
-    dockerfile: The path to a Dockerfile
-    dockerfile_contents: the raw contents of the Dockerfile to use for this build
+    ref: name for this image (e.g. 'myproj/backend' or 'myregistry/myproj/backend'). If this image will be used in a k8s resource(s), this ref must match the ``spec.container.image`` param for that resource(s).
+    context: path to use as the Docker build context.
+    build_args: build-time variables that are accessed like regular environment variables in the ``RUN`` instruction of the Dockerfile. See `the Docker Build Arg documentation <https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg>`_
+    dockerfile: path to the Dockerfile to build (may be absolute, or relative to the build context)
```

Comment:
i thought this was relative to the Tiltfile, not relative to the build context

Reply:
Ugh you're right, good catch.

Reply:
Hmmm maybe we want to rethink this? I'm think of both the docker cli and Docker Compose, which assume the Dockerfile is relative to the build context...

Reply:
I don't think the docker cli is at least:

```
> docker build go/src/github.com/windmilleng/servantes/snack/ -f Dockerfile
unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /Users/dan/Dockerfile: no such file or directory
```

Reply:
🤔 huh okay! (what if you don't pass -f tho?)

Reply:
this isn't true? I just tried:

```
docker build -f Dockerfile.docserver docs
```

definitely uses the Dockerfile in the current working directory, not relative to docs.

Reply:
then yes, it will look for `Dockerfile` in the root of the build context

Reply:
yeah if you pass `-f` it assumes the arg is relative to cwd but if you don't it looks for Dockerfile in the context... hmmm...

Reply:
Yep! And that's by design and documented behavior